cmake/flash/CMakeLists.txt

```diff
@@ -1,20 +1,64 @@
 # SPDX-License-Identifier: Apache-2.0
 
-add_custom_target(runner_yml_props_target)
-
 function(runner_yml_write content)
   # Append ${content}\n to a target property which is later evaluated as a
   # generator expression when writing the flash runner yaml file.
   # We define this function here to have access to the `flash` target.
 
   set_property(
-    TARGET         runner_yml_props_target
+    TARGET         runners_yaml_props_target
```

Comment:
Not related to this line, but would it be possible at the top of this file to list the format of the file ?
Or refer to a place where it can be found.

It may be confusing to some, that `runners_yaml_props_target` is not holding `property:value` pairs, but instead the content of a file to be written.

It's kind of misusing a target and property for this, so should be described well.

Reply:
> `runners_yaml_props_target` is not holding `property:value` pairs, but instead the content of a file to be written.

That's not true. The target named `runners_yaml_props_target` is holding property/value pairs. Its `yaml_contents` property has a value which is the YAML content of the file to be written.

> It's kind of misusing a target and property for this, so should be described well.

Is that still true, given the above?

Reply:
Also note that its `elf_file`, `hex_file`, and `bin_file` properties also hold their own separate values, due to this PR.

Reply:
> > `runners_yaml_props_target` is not holding `property:value` pairs, but instead the content of a file to be written.
> 
> That's not true. The target named `runners_yaml_props_target` is holding property/value pairs. Its `yaml_contents` property has a value which is the YAML content of the file to be written.

Right, `runners_yaml_props_target` != `yaml_contents`

> 
> > It's kind of misusing a target and property for this, so should be described well.
> 
> Is that still true, given the above?

No, it is completely correct usage.

Just to avoid confusing myself and others in the future, than I would suggest that the following text is added in top of this file:
```
The `runners_yaml_props_target` holds property:value pairs:
Property:        Description:
bin_file         Name of bin file generated by the build system
elf_file         Name of elf file generated by the build system
hex_file         Name of hex file generated by the build system
yaml_contents    Contents of runner file to be generated by the build system
Note: if if mcuboot and signing is enabled, the files above will be the signed versions.
```

Reply:
> Just to avoid confusing myself and others in the future, than I would suggest that the following text is added in top of this file:

Sure, good idea. I've added more details.